controllers/front/Payment.php

```diff
@@ -286,6 +287,17 @@ public function buildCCData($request, $payload)
             $request['paymentMethod']['holderName'] = $holderName;
         }
 
+        $shopperReference = $this->context->cart->id_customer;
+        if(!empty($shopperReference)) {
+            $request['shopperReference'] = $shopperReference;
+        }
+
+        //Oneclick data
+        if(!empty($payload['recurringDetailReference']) &&
+        $recurringDetailReference = $payload['recurringDetailReference']) {
+            $request['paymentMethod']['recurringDetailReference'] = $recurringDetailReference;
```

Comment:
```suggestion
            $request['paymentMethod']['recurringDetailReference'] = $payload['recurringDetailReference'];
```

Bit easier to read IMO but you can keep as it is if you think :)